### ISSUE TITLE
font scaling/rendering terribily broken

### ISSUE THREAD

**Issue by vasily-codefresh** [microsoft/terminal#4591]
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Windows Terminal version (if applicable):

Any other software?
```
Microsoft Windows NT 10.0.18362.0  
Windows Terminal (Preview)
Version: 0.9.433.0

# Steps to reproduce

* set display scaling to some custom setting. I use 130%
* launch Windows Terminal

# Expected behavior

fonts are scaled together with the rest of the system

# Actual behavior

fonts are scaled weirdly with wrong size and visible artefacts

compare those screenshots.

1. regular Windows console using Consolas 16pt. everything's fine here

<img src="https://ch3301files.storage.live.com/y4mvQqK8lX12c92wD2yKl8AWFCR4-OiImP9YaP5lHtm_4Ch-8VF1auHa30RaN7Zf37o13Co0qLbQ1cLcfSyHy5HI_mZgjDAD4oXJDdXMWUDsBC35Se2RnLgVndI_fX3GDyXD4BWmI68MDglG73nB6Izn4ZhkSyc2xHCMsVJPKCxbFQqwmD0-P0dtvly_I_XRZ7Vmept7qQXhOkvbo29JWEMVg/powershell.png?psid=1&width=822&height=166">

2. Windows Terminal using Consolas 16pt. somehow this 16pt is much larger

<img src="https://ch3301files.storage.live.com/y4mRt95zvzlVTVU1ssdoIhMNddwaRCKxr934xZ-_H3xDAcKCfdjg7HdysSwr2l2r_Ma3hpzT1Bci4o5gubOo8WzZCaHoC3GC9uEYcWOTBN1voN902x53KxRTieEHP1TrzSihUah8TB45lHH0L0skr6HbwTD7fHZMo4bmBCnLtcQLBT3KtzOq5O53bCFVwu_EQiPabpnAb95c99_hH705ooThA/wt16.png?psid=1&width=1274&height=257">

3. Windows Terminal using Consolas 11 pt (almost matches size in Windows console). notice broken fonts

<img src="https://ch3301files.storage.live.com/y4mFizZJYEVbhkeV8exgekvRcxudjIsm4JCT96bj-1NSkcgSSiRXoVccsmShMyPFeCQYMzxP0xMy7YHRVzFZ_25g74KefwoZFUHzgCiu9rRTzenhwPPTO1abAU-zvzXuqwX6CyG1vXNRDch5Mni4-rGCNg8p5np3AGtDkVZjctGFjwEspM3j_dABLpMDgzd9VdZPFfiItFantyjDE_w5JMw2g/wt11.png?psid=1&width=856&height=191">

font rendering has visible artifacts on almost all letters, very notable on `щ`, `й`, `з`, `б`, `р`, `ч` and even English `a` and `e`. the screenshot is a lossless PNG, that's how Windows Terminal display those in real life.

**Comment by j4james** [comment]
Note that the old console measures font sizes in pixels, not points (see issue #142). So that's why the sizes are so different.

I'm not sure about the font rendering artifacts - even zoomed in they look almost identical to me - but I'm probably just not a good judge of quality. Also there are definitely jpeg rendering artifacts. Your original image may have been a lossless PNG, but the github-hosted images are JPEG.

**Comment by vasily-codefresh** [comment]
@j4james good catch. please see original [uncompressed PNG](https://1drv.ms/u/s!AqSdRn_IUsGpga8V4D9Ej5RVHQx_Zw?e=b7s3Q1)

there a lot of artifacts. for example see the second line, first `о` in `Корпорация`

**Comment by j4james** [comment]
The new image still seems to be a jpeg, but now that I know where to look, I can definitely see what you mean about the artifacts. I wonder if it might be related to your graphics driver. I believe the old console uses GDI for rendering, while the new Windows Terminal uses DirectX, which would explain why you only see the issue in WT. That's just a wild guess though.

**Comment by vasily-codefresh** [comment]
dxgiag sees no problems.

you can try to reproduce this by setting custom scaling on Windows at 130% and rebooting.

**Comment by DHowett-MSFT** [comment]
I believe this is a /dupe of #4367. Agree?

**Comment by j4james** [comment]
@DHowett-MSFT Just to be clear, it's not a problem with the character heights - there are little rendering glitches to the right of some of the letters. I don't know if that makes it different from #4367 or not.

If you zoom in, it becomes more obvious where the problems are. For example, look at the little bump on the side of the second 'o' in 'Corporation' compared to the first 'o'.

![image](https://user-images.githubusercontent.com/4181424/74771609-5a29b700-5286-11ea-876d-239aebb84c7a.png)

I haven't been able to reproduce it myself though.

**Comment by DHowett-MSFT** [comment]
_huh_

**Comment by zadjii-msft** [comment]
I definitely see a repro of this with 130% scaling, font size 9, grayscale or cleartype AA, "Fira Code" as my font. Also repros at size 9 with "Hack" and "Cascadia Code"

When it repros for me, it's happening in the following columns:
```
Microsoft Windows [Version 10.0.19604.1002]
(c) 2020 Microsoft Corporation. All rights reserved.
                                            
migrie@MIGRIE-SLAPTOP>oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo
oooooooooooooooooooooooooooooooooooooooooooooo
      ^             ^             ^               
      7             21            35
```

**Comment by zadjii-msft** [comment]
Okay, I'm punting this till the next release, because it's a fairly uncommon scenario, that peculiarly doesn't repro with other DPI scalings at the standard 100%/125%/150% (etc.) scalings. 

![image](https://user-images.githubusercontent.com/18356694/79134551-825c0080-7d73-11ea-8621-5e008af60087.png)

I tried creating a branch where I used a `float` to store our scaling and DPI. This did seem to make the problem far less prevalent, but didn't entirely resolve the issue. If you look at the above screenshot, you'll see the artifact will be visible on all characters in the column at a specific pixel distance from the left of the window. In this screenshot, it's the right side of all the "WOQ" column of characters.

If you decrease the font size, the artifact continues to occur at that specific pixel column, not that character column. 

Maybe there's somewhere else in the DX renderer that we're doing more `int` math instead of `float` math?

Note that it also repros with grayscale AA:
![image](https://user-images.githubusercontent.com/18356694/79135151-86d4e900-7d74-11ea-907b-dbaf16cce8de.png)

**Comment by LinoBarreca** [comment]
Until I had this problem i underestimated the importance of a clean font rendering. The problem is so bad on my side that I can not concentrate when using the terminal...and i swear i don't suffer from OCD. The glitches are not minor. The lines are deformed both in height and width. For sample pictures look at #8533 (sorry for the duplicate, i searched for it and i found only a bug in which they were referring to a problem with intel cards and acceleration but that wasn't my case)

**Comment by adhadse** [comment]
Found same issue happening on Windows Terminal window for Ubuntu at font 12, 'JetBrains Mono' on Version: 1.9.1942.0.

My screen scaling is set to 110%. Any type of anti-aliasing other than default "Grayscale" gets even worse. 

Check the ticked text and compare that to usual text:
- Like the two 'n' ticked in blue
- The two 'i' ticked in yellow
- The two '5' ticked in maroon.
![image](https://user-images.githubusercontent.com/56764399/132659991-9f3990a5-c677-4728-a8c2-b4d30439e878.png)

**Comment by kXborg** [comment]
I don't know how you guys are closing the issue. I don't see any proper fixes.
Setting display scaling back to default 100% does improve the rendering a little but this too is not very good. Take a look at the screenshot below. Left side is of Terminal and Right side is of Default Powershell. Both are using same text and scale. Please provide fixes. The texts are not beautiful at all (in Terminal).
![Terminal-Powershell](https://user-images.githubusercontent.com/55614432/140271398-83eddcfd-4115-41ea-828e-d61d2dce31d5.jpg)

**Comment by kXborg** [comment]
This is the zoomed view. Zooming out makes me feel like I am using a 640x480 pixel display.

**Comment by zadjii-msft** [comment]
> I don't know how you guys are closing the issue. I don't see any proper fixes.

We're not? I'm simply combining threads reporting the same issue to a single thread, for easy tracking. Custom display scaling sizes, like 130%, or 118% cause a LOT of visual artifacts, that we're using this thread to track resolving. 


@kXborg  From the settings you sent me in #11679, you have text antialiasing turned off:

```json
{
    "altGrAliasing": true,
    "antialiasingMode": "aliased",
    "colorScheme": "Campbell Powershell",
    "commandline": "powershell.exe",
    "font": 
    {
        "face": "Courier New"
    },
    "guid": "{61c54bbd-c2c6-5271-96e7-009a87ff44bf}",
    "hidden": false,
    "name": "Windows PowerShell"
},
```

Getting rid of the `"antialiasingMode": "aliased",` will probably help you.

**Comment by kXborg** [comment]
@zadjii-msft, <br> Hi, it fixed the issue. Thanks a lot, Mike

**Comment by michaelbing3shape** [comment]
I would like to vote for this to be fixed. It looks awful even with grayscale AA. Regular PowerShell looks fine. 110% scaling. I don't have problems with any other apps.

![image](https://user-images.githubusercontent.com/69194362/146359936-ec3d2a19-b648-4b6a-b0b5-438a7e184546.png)
![image](https://user-images.githubusercontent.com/69194362/146359956-04b2e8bd-6ed1-45c2-bb58-898516e4bfdd.png)

**Comment by zadjii-msft** [comment]
Hey random question: Is this still happening to anyone with the latest iteration of the Atlas engine? I wonder if we just fixed this as a drive-by.
